packages/botbuilder-filter-middleware/src/filterMiddleware.ts

```diff
@@ -0,0 +1,26 @@
+import { Activity, Middleware, ConversationResourceResponse, MiddlewareSet } from 'botbuilder';
+
+export function filterMiddleware(predicate: (context: BotContext) => boolean, ...middlewares: Middleware[]): MiddlewareSet {
+    let middlewareSet = new MiddlewareSet();
+
+    middlewares.forEach(middleware => {
+        middlewareSet.use({
+            contextCreated: (context: BotContext, next: () => Promise<void>) => {
+                return (predicate(context) && middleware.contextCreated)
```

Comment:
>predicate(context)  [](start = 24, length = 19)

With this approach you'll end up calling your predicate() once per piece of middleware added. So if I ad 5 pieces of middleware I'll call the predicate 5 times. Would it be better to just call it once?